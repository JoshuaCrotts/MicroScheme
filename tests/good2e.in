
// mixed types in binary op - cast the char to an int

void main() {
    char ca = 1;
    char cb = 2;
    int ia = 1;
    int ib = 2;
    char cc;
    int  ic;

    ic = ia * cb;
    printd(ic);
}

