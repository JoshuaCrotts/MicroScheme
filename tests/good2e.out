void SEQ (
  void FNDEF main void () (
    void SEQ (
      void DECL ca (char) = 1,
      void DECL cb (char) = 2,
      void DECL ia (int) = 1,
      void DECL ib (int) = 2,
      void DECL cc (char),
      void DECL ic (int),
      int ASN (
        int ID ic,
        int BINOP('*') (
          int ID ia,
          int CAST (
            char ID cb))),
      void FNCALL printd (
        int ID ic))))
